packages/browser-wallet/src/shared/utils/token-helpers.ts

```diff
@@ -0,0 +1,123 @@
+import { Buffer } from 'buffer/';
+import { InstanceInfo, JsonRpcClient } from '@concordium/web-sdk';
+import { TokenMetadata } from '@shared/storage/types';
+
+export interface TokenIdAndMetadata {
+    id: string;
+    metadataLink: string;
+    metadata: TokenMetadata;
+}
+
+export interface ContractDetails {
+    contractName: string;
+    index: bigint;
+    subindex: bigint;
+}
+
+/**
+ * Returns a buffer containing the parameter used to check whether a smart contract is CIS-2 compliant. (Using the CIS-0 view function .supports)
+ */
+export function getCIS2Identifier(): Buffer {
+    const buf = Buffer.alloc(8);
+    buf.writeInt16LE(1, 0);
+    buf.writeInt8(5, 2);
+    buf.write('CIS-2', 3, 5, 'ASCII');
+    return buf;
+}
+
+/**
+ * Returns a buffer containing the parameters used for the CIS-2 view function .tokenMetadata, for the given token id.
+ */
+export function getMetadataParameter(id: string): Buffer {
+    const lengths = Buffer.alloc(3);
+    const idBuf = Buffer.from(id, 'hex');
+    lengths.writeInt16LE(1, 0);
+    lengths.writeInt8(idBuf.length, 2);
+    return Buffer.concat([lengths, idBuf]);
+}
+
+/**
+ * Confirms that the given smart contract instance is CIS-2 compliant
+ */
+export async function confirmCIS2Contract(
+    client: JsonRpcClient,
+    instanceInfo: InstanceInfo,
+    { contractName, index, subindex }: ContractDetails
+): Promise<string | undefined> {
+    if (!instanceInfo.methods.includes(`${contractName}.supports`)) {
+        return 'Chosen contract does not support CIS-0';
+    }
+    if (
+        !(
+            instanceInfo.methods.includes(`${contractName}.balanceOf`) &&
+            instanceInfo.methods.includes(`${contractName}.transfer`) &&
+            instanceInfo.methods.includes(`${contractName}.tokenMetadata`)
+        )
```

Comment:
Would it maybe be nicer to define the required functions in a constant outside, and check that `instanceInfo.methods` include all of them?

```ts
const requried = ['balanceOf', ...];
instanceInfo.methods.every(required.map(r => `${contractName}.${r}).includes);
```